core/watcher/eventsource/value.go

```diff
@@ -147,9 +178,65 @@ func (w *ValueWatcher) drainInitialEvent(in <-chan []changestream.ChangeEvent) {
 	select {
 	case _, ok := <-in:
 		if !ok {
-			w.logger.Debugf(context.TODO(), "change channel closed for %q; terminating watcher", w.namespace)
+			w.logger.Debugf(context.Background(), "change channel closed; terminating watcher")
```

Comment:
Just after `drainInitialEvent` is called we set up a context. Can we tweak things and pass that in and use it.

Reply:
Sure.